tenants-certified-attributes-updater/src/main/scala/it/pagopa/interop/tenantscertifiedattributesupdater/util/AttributeInfo.scala

```diff
@@ -9,21 +14,46 @@ final case class AttributeInfo(origin: String, code: String, revocationTimestamp
 
 object AttributeInfo {
   def addRevocationTimeStamp(
-    attribute: PersistentTenantAttribute,
     attributesIndex: Map[UUID, AttributeInfo]
-  ): Option[AttributeInfo] =
-    attribute match {
-      case PersistentCertifiedAttribute(id, _, revocationTimestamp) =>
-        attributesIndex.get(id).map(_.copy(revocationTimestamp = revocationTimestamp))
-      case _                                                        => attributesIndex.get(attribute.id)
-    }
+  ): PersistentTenantAttribute => Option[AttributeInfo] = {
+    case PersistentCertifiedAttribute(id, _, revocationTimestamp) =>
+      attributesIndex.get(id).map(_.copy(revocationTimestamp = revocationTimestamp))
+    case attribute                                                => attributesIndex.get(attribute.id)
+  }
+
+  def canBeActivated(fromTenant: Map[PersistentExternalId, List[AttributeInfo]]): TenantSeed => Boolean =
+    tenantSeed => !canNotBeActivated(fromTenant, tenantSeed)
 
-  def stillExistsInTenant(attributeFromRegistry: AttributeInfo): List[AttributeInfo] => Boolean =
-    attributesFromTenant =>
-      attributesFromTenant.exists(attributeFromTenant =>
-        attributeFromTenant.code == attributeFromRegistry.code &&
-          attributeFromTenant.origin == attributeFromRegistry.origin &&
-          attributeFromTenant.revocationTimestamp.isEmpty
+  private def canNotBeActivated(
+    fromTenant: Map[PersistentExternalId, List[AttributeInfo]],
+    tenantSeed: TenantSeed
+  ): Boolean =
+    fromTenant
+      .get(tenantSeed.id.toPersistentExternalId)
+      .exists(attributesFromTenant => // check if exist in tenant
+        tenantSeed.attributesInfo.isEmpty || tenantSeed.attributesInfo.exists(attributeFromRegistry =>
+          attributesFromTenant.exists { attributeFromTenant =>
+            attributeFromTenant.code == attributeFromRegistry.code &&
+            attributeFromTenant.origin == attributeFromRegistry.origin &&
+            attributeFromTenant.revocationTimestamp.isEmpty
+          }
+        )
       )
 
+  def isRevocable(fromRegistry: List[TenantSeed]): (PersistentExternalId, AttributeInfo) => Boolean =
+    (tenantId, attributeFromTenant) => !isNotRevocable(fromRegistry, tenantId, attributeFromTenant)
+
+  private def isNotRevocable(
+    fromRegistry: List[TenantSeed],
+    tenantId: PersistentExternalId,
+    attributeFromTenant: AttributeInfo
+  ): Boolean =
+    fromRegistry.exists { tenantSeed => // check if exists in tenant
+      tenantSeed.id.toPersistentExternalId == tenantId && tenantSeed.attributesInfo.isEmpty || tenantSeed.attributesInfo
+        .exists(attributeFromRegistry =>
+          attributeFromTenant.code == attributeFromRegistry.code &&
+            attributeFromTenant.origin == attributeFromRegistry.origin
+        )
```

Comment:
Maybe I got it wrong, but is it possible that this should be
```suggestion
      tenantSeed.id.toPersistentExternalId == tenantId && (tenantSeed.attributesInfo.isEmpty || tenantSeed.attributesInfo
        .exists(attributeFromRegistry =>
          attributeFromTenant.code == attributeFromRegistry.code &&
            attributeFromTenant.origin == attributeFromRegistry.origin
        )
      )
```